### PR TITLE
Implement Duplicate column (CSV rule)

### LINKED ISSUE #129: [REQUEST] Implement Duplicate column (CSV rule)
**Is your feature request related to a problem? Please describe.**
Duplicate column is a CSV rule implemented in Google Python validator and featured in Google Type Error as TYPE_CSV_DUPLICATE_COLUMN_NAME.

**Describe the solution you'd like**
A column must not be duplicated in a GTFS file.

**Describe alternatives you've considered**
**Additional context**
Line 22 in Error support priority https://docs.google.com/spreadsheets/d/1vqe6wq7ctqk1EhYkgtZ0_TbcQ91vccfs2daSjn20BLE/edit#gid=0

### PR DESCRIPTION
**Pull request by fabrice-v**
**Summary:**

close #129 

This PR adds a rule to check if a column is duplicated in an input file.
A `DuplicatedHeaderNotice` is built for each duplicate found, even for out of spec fields.

**Expected behavior:** 

[MTBA_duplicated_agency_name.zip](https://github.com/MobilityData/gtfs-validator/files/4997168/MTBA_duplicated_agency_name.zip)

image

Please make sure these boxes are checked before submitting your pull request - thanks!

- [x] Run the unit tests with `gradle test` to make sure you didn't break anything
- [ ] Format the title like "Fix #<issue_number> - <short description of fix and changes>" (for example - "Fix #1111 - Check for null value before using field")
- [x] Linked all relevant issues
- [x] Include screenshot(s) showing how this pull request works and fixes the issue(s)

### CODE REVIEW

File: RULES.md [hunk 1]
```diff
@@ -222,15 +229,27 @@ Any other combination raise this error
...
 Field `parent_station` must be empty when `location_type` is 2
 
 #### References:
 * [stops.txt specification](http://gtfs.org/reference/static/#stopstxt)
 
+<a name="E043"/>
+
+### E043 - Duplicated field
+
+Some file contains the same column name twice
```
Comment (barbeau) [inline RULES.md:243]:
```suggestion
A file cannot contain the same header value twice (i.e., duplicated column of data).
```

File: domain/src/main/java/org/mobilitydata/gtfsvalidator/domain/entity/notice/error/MissingHeaderNotice.java [hunk 2]
```diff
@@ -25,7 +25,7 @@
 
     public MissingHeaderNotice(final String filename, final String missingHeaderName) {
         super(filename, E_001,
-                "Field name not defined in the GTFS specification",
```
Comment (barbeau) [inline domain/src/main/java/org/mobilitydata/gtfsvalidator/domain/entity/notice/error/MissingHeaderNotice.java:28]: Just flagging unrelated changes - not sure if they are intentional. A few others exist in this PR as well.

File: adapter/repository/in-memory-simple/src/main/java/org/mobilitydata/gtfsvalidator/db/InMemoryRawFileRepository.java [hunk 3]
```diff
@@ -69,15 +69,15 @@ public RawFileInfo create(RawFileInfo fileInfo) {
...
+        Collection<String> toReturn = new ArrayList<>();
 
         try {
             ((CsvSchema) (mapper.readerFor(Map.class)
                     .with(schema)
                     .readValues(csvFile).getParser().getSchema())).iterator().forEachRemaining(column -> toReturn.add(column.getName()));
         } catch (IOException e) {
             //TODO: this should go back up to use case level so it can be properly reported
-            return Collections.emptySet();
+            return Collections.emptyList();
```
Comment (barbeau) [inline adapter/repository/in-memory-simple/src/main/java/org/mobilitydata/gtfsvalidator/db/InMemoryRawFileRepository.java:80]: I'd suggest adding a note to the Javadoc that says that the returned list will contain all headers present in the CSV file, including any duplicates.

Comment (barbeau) [reply adapter/repository/in-memory-simple/src/main/java/org/mobilitydata/gtfsvalidator/db/InMemoryRawFileRepository.java:80]: Actually, thinking more, given that this actually affects program correctness, it's probably better to change the return value explicitly to a `List` (in addition to documenting in the Javadoc). For example, you wouldn't want someone to change the implementation of the method internally back to a `Set` in the future, thinking that it still satisfies the `Collection` interface, because E43 would never find any duplicate columns. Changing the return type, IMHO, is a more explicit statement that there is a reason why a `List` is used here.

File: domain/src/main/java/org/mobilitydata/gtfsvalidator/domain/entity/notice/error/MissingHeaderNotice.java (hunk 2 above)
Comment (fabrice-v) [reply domain/src/main/java/org/mobilitydata/gtfsvalidator/domain/entity/notice/error/MissingHeaderNotice.java:28]: Ok yes. This one is intentional  (the message is wrong)
